docker-app/qfieldcloud/core/models.py

```diff
@@ -685,6 +705,15 @@ def clean(self) -> None:
 
         return super().clean()
 
+    @transaction.atomic
+    def delete(self, *args, **kwargs) -> None:
+        super().delete(*args, **kwargs)
+
+        TeamMember.objects.filter(
+            team__team_organization=self.organization,
+            member=self.member,
+        ).delete()
```

Comment:
I am not 100% sure if it is ok to mark the whole delete method as in transaction like this, but IMO should be in transaction in any case.